modules/proxy-certificates.adoc

```diff
@@ -0,0 +1,169 @@
+    Custom CA certificate bundle.
+    -----END CERTIFICATE-----
+----
+
+[discrete]
+== Managing proxy certificates during installation
+
+The `additionalTrustBundle` value of the installer configuration is used to
+specify any proxy-trusted CA certificates during installation. For example:
+
+----
+$ cat install-config.yaml
+. . .
+proxy:
+  httpProxy: http://<HTTP_PROXY>
+  httpsProxy: https://<HTTPS_PROXY>
+additionalTrustBundle: |
+    -----BEGIN CERTIFICATE-----
+   <MY_HTTPS_PROXY_TRUSTED_CA_CERT>
+    -----END CERTIFICATE-----
+. . .
+----
+
+[discrete]
+== Location
+
+The user-provided trust bundle is represented as a ConfigMap. The ConfigMap is
+mounted into the file system of platform components that make egress HTTPS
+calls. Typically, Operators mount the ConfigMap to
+`/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`, but this is not required by
+the proxy. A proxy can modify or inspect the HTTPS connection. In either case,
+the proxy must generate and sign a new certificate for the connection.
+
+Complete proxy support means connecting to the specified proxy and trusting any
+signatures it has generated. Therefore, it is necessary to let the user specify
+a trusted root, such that any certificate chain connected to that trusted root
+is also trusted.
+
+If using the RHCOS trust bundle, place CA certificates in
+`/etc/pki/ca-trust/source/anchors`.
+
+See
+link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/using-shared-system-certificates_security-hardening[Using
+shared system certificates] in the Red Hat Enterprise Linux documentation for
+more information.
+
+[discrete]
+== Expiration
+
+The user sets the expiration term of the user-provided trust bundle.
+
+The default expiration term is defined by the CA certificate itself. It is up to
+the CA administrator to configure this for the certificate before it can be used
+by {product-title} or {op-system}.
+
+[NOTE]
+====
+Red Hat does not monitor for when CAs expire. However, due to the long life of
+CAs, this is generally not an issue. However, you might need to periodically
+update the trust bundle.
+====
+
+[discrete]
+== Services
+
+By default, all platform components that make egress HTTPS calls will use the
+{op-system} trust bundle. If `trustedCA` is defined, it will also be used.
+
+Any service that is running on the {op-system} node is able to use the trust bundle of
+the node.
+
+[discrete]
+== Management
+
+These certificates are managed by the system and not the user.
+
+[discrete]
+== Customization
+
+Updating the user-provided trust bundle consists of either:
+
+* updating the PEM-encoded certificates in the ConfigMap referenced by
+`trustedCA,` or
+* creating a ConfigMap in the namespace `openshift-config` that contains the new
+trust bundle and updating `trustedCA` to reference the name of the new
+ConfigMap.
+
+The mechanism for writing CA certificates to the {op-system} trust bundle is exactly
```

Comment:
I'm confused by the rest of the text in this section. This looks like how MCO implements proxy trustedCA. cc: @runcom

The mechanism operators use for writing the trust bundle consists of:
-  The operator requests trust bundle injection by creating a `ConfigMap` in the operator's namespace with label `config.openshift.io/inject-trusted-cabundle: "true"`. Here's an example of the Ingress Operator:
```
apiVersion: v1
kind: ConfigMap
metadata:
  annotations:
    release.openshift.io/create-only: "true"
  labels:
    config.openshift.io/inject-trusted-cabundle: "true"
  name: trusted-ca
  namespace: openshift-ingress-operator
```
- Cluster Network Operator injects the trusted ca bundle into this `ConfigMap`:
```
kind: ConfigMap
metadata:
  annotations:
    release.openshift.io/create-only: "true"
  labels:
    config.openshift.io/inject-trusted-cabundle: "true"
  name: trusted-ca
  namespace: openshift-ingress-operator
apiVersion: v1
data:
  ca-bundle.crt: |
    <PEM_ENCODED_TRUSTED_CA_CERTS>
```
`ca-bundle.crt` contains either the RHCOS trust bundle or the merged RHCOS/user-provided bundle.

- If the operator makes egress requests, it will typically mount this `ConfigMap` to `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`. If the operand makes egress requests, the operator will plumb the contents of `ca-bundle.crt` into the operand's trust store, typically `/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem`.
- The operator watches the `ConfigMap` for changes and updates the trust bundle accordingly.